api/all/src/main/java/io/opentelemetry/api/internal/ConfigUtil.java

```diff
@@ -33,10 +36,18 @@ private ConfigUtil() {}
    */
   public static String getString(String key, String defaultValue) {
     String normalizedKey = normalizePropertyKey(key);
+    Set<Map.Entry<String, String>> properties =
+        System.getProperties().stringPropertyNames().stream()
+            .map(
+                propertyName ->
+                    new AbstractMap.SimpleEntry<>(propertyName, System.getProperty(propertyName)))
+            .filter(entry -> entry.getKey() != null)
+            .collect(Collectors.<Map.Entry<String, String>>toSet());
+
     String systemProperty =
-        System.getProperties().entrySet().stream()
-            .filter(entry -> normalizedKey.equals(normalizePropertyKey(entry.getKey().toString())))
-            .map(entry -> entry.getValue().toString())
+        properties.stream()
+            .filter(entry -> normalizedKey.equals(normalizePropertyKey(entry.getKey())))
+            .map(Map.Entry::getValue)
```

Comment:
I think you could apply the filter directly on the `stringPropertyNames` (without creating the temporary map)

Reply:
You are completely right! Just changed it. Hope that's what you meant :).